lib/openstudio-standards/standards/necb/NECB2011/necb_2011.rb

```diff
@@ -799,9 +822,17 @@ def apply_kiva_foundation(model)
             zone_kiva_models << kiva_model
           end
           # Kiva model only works with standard materials. Replace constructions massless materials with standard ones.
-          replace_massless_material_with_std_material(model,gfloor)
-          gfloor.setOutsideBoundaryCondition('Foundation')
-          gfloor.setAdjacentFoundation(zone_kiva_models.last)
+          if massless
+            c = gfloor.construction.get.to_LayeredConstruction.get
+            gfloor.setOutsideBoundaryCondition('Foundation')
+            gfloor.setAdjacentFoundation(zone_kiva_models.last)
+            gfloor.setConstruction(c)
+          else
+            replace_massless_material_with_std_material(model,gfloor)
```

Comment:
_Foundation_-facing surfaces have their construction reset in _replace_massless_material_with_std_material_. The optional building _structure_-based construction set generation avoids _massless_ materials altogether.